src/components/layout/MainLayout/MainLayout.js

```diff
@@ -5,17 +5,41 @@ import Header from '../Header/Header';
 import Footer from '../Footer/Footer';
 import ProductsCompare from '../ProductsCompare/ProductsCompareContainer';
 
-const MainLayout = ({ children }) => (
-  <div>
-    <Header />
-    {children}
-    <Footer />
-    <ProductsCompare />
-  </div>
-);
+class MainLayout extends React.Component {
+  checkMode = val => {
+    return val <= 767 ? 'mobile' : val <= 1024 ? 'tablet' : 'desktop';
+  };
+
+  componentWillMount() {
+    const { setRwdMode } = this.props;
+    window.addEventListener('resize', e => {
+      const mode = this.checkMode(e.target.window.innerWidth);
+      setRwdMode(mode);
+      return null;
+    });
+  }
+
+  componentDidMount() {
+    const event = new Event('resize');
+    window.dispatchEvent(event);
```

Comment:
Po co jest ten kawałek kodu?

Brakuje removeEventListener

Reply:
po dodaniu listnera robie dispatch eventu resize by komponent dostał odpowiednie propsy, bez tego nie pokazuje kart, bo nie argumentu.

Reply:
No to funkcję która ma się wykonać po resize zapisz jako osobną funkcję, wywołaj ją przy wykryciu eventu resize oraz bezpośrednio tutaj ;)

Reply:
można :) wywaliłem dispatch eventu i wywołałem z propsów:
componentDidMount() {
    const { setRwdMode } = this.props;
    setRwdMode(this.checkMode(window.innerWidth));
  }